BridgeCareApp/VuejsApp/src/components/DetailedReport.vue

```diff
@@ -6,193 +6,206 @@
-        warning: IAlert
-
+        /**
+         * Component has been mounted
+         */
         mounted() {
-            this.downloadProgress = true
-            this.loading = true
-
-            this.$store.dispatch({
-                type: 'getNetworks'
-            }).then(() => {
-                this.downloadProgress = false
-                this.loading = false
-                this.networks = this.$store.getters.networks as INetwork[]
-            }).catch((error) => {
-                this.downloadProgress = false
-                this.loading = false
-                console.log(error)
-            })
+            // dispatch action to get networks
+            this.setIsBusyAction({isBusy: true});
+            this.getNetworksAction().then(() =>
+                this.setIsBusyAction({isBusy: false})
+            ).catch((error: any) => {
+                this.setIsBusyAction({isBusy: false});
+                console.log(error);
+            });
         }
 
-        getSimulations(id: number) {
-            this.networkId = id
-            let details = this.networks.find(t => t.networkId === id) as INetwork
-            this.networkName = details.networkName
-
-            this.$store.dispatch({
-                type: 'getSimulations',
-                id: id
-            }).then(() => {
-                this.simulations = this.$store.getters.simulations as Simulation[]
-                    this.disableSimulationDropDown = false
-                }).catch((error) => {
-                    console.log(error)
-                })
+        /**
+         * A network has been selected
+         * @param networkId The selected network id
+         */
+        onSelectNetwork(networkId: number) {
+            this.networkId = networkId;
+            // @ts-ignore
+            const selectedNetwork: Network = this.networks.find(t => t.networkId === networkId);
+            this.networkName = hasValue(selectedNetwork) ? selectedNetwork.networkName : "";
+            // dispatch action to get simulations
+            this.setIsBusyAction({isBusy: true});
+            this.getSimulationsAction({networkId: networkId}).then(() =>
+                this.setIsBusyAction({isBusy: false})
+            ).catch((error: any) => {
+                this.setIsBusyAction({isBusy: false});
+                console.log(error);
+            });
         }
 
-        getSimulationId(id: any) {
-            this.simulationId = id
-            let detail = this.simulations.find(_ => _.simulationId == id) as Simulation
-            this.simulationName = detail.simulationName
-            this.disableDownloadReport = false
-            this.disableRunSimulation = false
+        /**
+         * A simulation has been selected
+         * @param simulationId The selected simulation id
+         */
+        onSelectSimulation(simulationId: number) {
+            this.simulationId = simulationId;
+            // @ts-ignore
```

Comment:
Sometimes typescript is a little too strict. Here the output from running 'npm run serve' to start the project kept complaining about a type of Network cannot be assigned undefined, but this is why I have the ternary checking if the selectedNetwork was actually found, otherwise setting the networkName to an empty string.